swift/apple/FirezoneKit/Sources/FirezoneKit/Helpers/SemanticVersion.swift

```diff
@@ -3,6 +3,7 @@
 //  (c) 2024 Firezone, Inc.
 //  LICENSE: Apache-2.0
 //
+// TODO: This file can be deleted once most users have upgraded past 1.4.14
```

Comment:
Or we say that migrations only work for N versions but not beyond that?

Reply:
Well this isn't user-facing, so as long as users land on a patch release

It's also not the most critical thing to preserve since it's just storing which update versions they've dismissed.

Also, we don't even need this class because String has native support for comparing version strings 🙃. I originally removed that in this PR but decided to revert to keep this better scoped.